python/langsmith/_internal/otel/_otel_exporter.py

```diff
@@ -163,7 +180,8 @@ def export_batch(
         Args:
             operations: List of serialized run operations to export.
         """
-        # Create a dictionary mapping run IDs to their operations
+        # Proactive cleanup of expired and excess spans before new operations
+        self._cleanup_stale_spans()
```

Comment:
This seems to iterate over all spans in memory, not sure we want to do that on every batch?

Could we somehow track the size of all in-memory spans and cleanup if it goes above a certain point?